src/mongo/util/processinfo_win32.cpp

```diff
@@ -116,6 +117,73 @@ namespace mongo {
 #endif
     }
 
+    bool getFileVersion(const char *filePath, DWORD &fileVersionMS, DWORD &fileVersionLS) {
+        DWORD verSize = GetFileVersionInfoSizeA(filePath, NULL);
+        if (verSize == 0) {
+            DWORD gle = GetLastError();
+            warning() << "GetFileVersionInfoSizeA on " << filePath << " failed with " << errnoWithDescription(gle);
+            return false;
+        }
+
+        boost::scoped_array<char> verData(new char[verSize]);
+        if (GetFileVersionInfoA(filePath, NULL, verSize, verData.get()) == 0) {
+            DWORD gle = GetLastError();
+            warning() << "GetFileVersionInfoSizeA on " << filePath << " failed with " << errnoWithDescription(gle);
+            return false;
+        }
+
+        UINT size;
+        LPBYTE lpBuffer;
+        if (VerQueryValueA(verData.get(), "\\", (LPVOID *)&lpBuffer, &size) == 0) {
+            DWORD gle = GetLastError();
+            warning() << "VerQueryValueA on " << filePath << " failed with " << errnoWithDescription(gle);
+            return false;
+        }
+    
+        if (size != sizeof(VS_FIXEDFILEINFO)) {
+            warning() << "VerQueryValueA on " << filePath << " returned structure with unexpected size";
+            return false;
+        }
+
+        VS_FIXEDFILEINFO *verInfo = (VS_FIXEDFILEINFO *)lpBuffer;
+        fileVersionMS = verInfo->dwFileVersionMS;
+        fileVersionLS = verInfo->dwFileVersionLS;
+        return true;
+    }
+
+    // If the version of the ntfs.sys driver shows that the KB2731284 hotfix or a later update
+    // is installed, zeroing out data files is unnecessary. The file version numbers used below
+    // are taken from the Hotfix File Information at http://support.microsoft.com/kb/2731284.
+    bool isKB2731284OrLaterUpdateInstalled() {
+        char systemRoot[MAX_PATH];
+        if (FAILED(SHGetSpecialFolderPathA(NULL, systemRoot, CSIDL_SYSTEM, FALSE))) {
+            DWORD gle = GetLastError();
+            warning() << "SHGetSpecialFolderPathA on CSIDL_SYSTEM failed with " << errnoWithDescription(gle);
+            return false;
+        }
+
+        string ntfsDotSysPath = systemRoot;
+        ntfsDotSysPath.append("\\drivers\\ntfs.sys");
```

Comment:
Do we have a guarantee that the returned path will not have a trailing backslash?  If not, then we should check for it.